task/sast-coverity-buildless-check/0.1/sast-coverity-buildless-check.yaml

```diff
@@ -0,0 +1,189 @@
+      # the cluster will set imagePullPolicy to IfNotPresent
+      workingDir: $(workspaces.workspace.path)/hacbs/$(context.task.name)
+      volumeMounts:
+        - name: cov-license
+          mountPath: "/etc/secrets"
+          readOnly: true
+      env:
+        - name: COV_ANALYZE_ARGS
+          value: $(params.COV_ANALYZE_ARGS)
+        - name: COV_CAPTURE_ARGS
+          value: $(params.COV_CAPTURE_ARGS)
+        - name: KFP_GIT_URL
+          value: $(params.KFP_GIT_URL)
+        - name: SEVERITY_THRESHOLD
+          value: $(params.SEVERITY_THRESHOLD)
+      script: |
+        #!/usr/bin/env bash
+        set -eo pipefail
+
+        echo 'Starting Coverity buildless scan'
+        SOURCE_CODE_DIR=$(workspaces.workspace.path)
+        COVERITY_DIR=/tmp/cov-scan/cov
+        COVERITY_RESULTS_FILE=coverity-buildless-results.js
+        COV_LICENSE_PATH=/etc/secrets/cov-license
+        # Installing Coverity licen
+        cp /etc/secrets/cov-license /opt/coverity/bin/license.dat
+        # Install dependencies
+        dnf copr enable copr.devel.redhat.com/@osh/csdiff -y
+        dnf install -y csdiff
+        if test -z "$(find /opt -name 'license.dat')"; then
+          echo "No license file for Coverity was detected. Exiting..."
+          exit 1
+        fi
+        
+        # Installation of Red Hat certificates for cloning Red Hat internal repositories
+        curl -sS https://certs.corp.redhat.com/certs/2015-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2015-RH-IT-Root-CA.crt
+        curl -sS https://certs.corp.redhat.com/certs/2022-IT-Root-CA.pem > /etc/pki/ca-trust/source/anchors/2022-IT-Root-CA.pem
+        update-ca-trust
+        
+        # Create configuration file for coverity buildless scan. New Coverity CLI only uses configuration files
+        cat > $SOURCE_CODE_DIR/coverity.yml << EOF
+        capture:
+          build-command-inference: false
+        analyze:    
+            checkers:
+              c-family-security: true        
+              concurrency: true
+              checker-config:            
+                hardcoded_credentials:
+                  enabled: true    
+            cov-analyze-args:
+              - '--include-java'
+              - '--spotbugs-max-mem=4096'
+        EOF
+        
+        # Captureless scan
+        env COV_HOST=konflux HOME=/tmp/cov-scan/home /opt/coverity/bin/coverity capture $COV_CAPTURE_ARGS --project-dir $SOURCE_CODE_DIR --dir $COVERITY_DIR
+        COV_CAPTURE_EXIT_CODE=$?
+        
+        if [[ "$COV_CAPTURE_EXIT_CODE" -eq 0 ]]; then
+            echo "Coverity capture scan finished succesfully"
+        else
+            echo "Coverity capture command failed. Exiting..."
+            exit 1
+        fi
+        
+        # Analysis phase
+        /opt/coverity/bin/cov-manage-emit --dir $COVERITY_DIR reset-host-name
+        /opt/coverity/bin/coverity analyze $COV_ANALYZE_ARGS --project-dir $SOURCE_CODE_DIR --dir=$COVERITY_DIR
```

Comment:
I do not think it makes sense to pass `$COV_ANALYZE_ARGS` to `coverity analyze`.  The command does not take any arguments that would be useful to set for end users.  It would not be possible to override the default args.  It would not be possible to exclude vendor directories from scanning, etc.  Converting the string to the appropriate section of `coverity.yml` in the shell script would be problematic.  Let's just use `cov-analyze` instead for now to not over-complicate things.  As far as I know, the `cov-analyze` command is not yet deprecated and it should be compatible with the new CLI.

Reply:
Removed `$COV_ANALYZE_ARGS` from` coverity analyze` and switched back to the `cov-analyze` CLI

Reply:
Sorry if I was unclear.  The only reason for using `cov-analyze`  was to make it possible to easily customize the scanning, e.g., to:
- override the default args, or
- exclude vendor directories from scanning

I did not meant to remove the `COV_ANALYZE_ARGS` parameter from the task.

Btw. I cannot find `--include-java` in the documentation.  I guess the option was deprecated long time ago.

Reply:
Gotcha. I thought you were referring to two different problems. Modified to use the parameter and using `cov-analyze`

Regarding the `--include-java` . I also could not find any reference while doing the migration to the `coverity.yml` file. Should we leave it as Coverity is not complianing about it and remove it in future version? Or should we remove it now?

Reply:
I was unable to find the answer to your question in the documentation, so I have asked Coverity Support about it.

Reply:
... and got an answer:  The `--include-java` option does not have any effect since Coverity 7.7.0, which was released 9 years ago.  We can safely remove it.

Reply:
Thanks for researching it. I have removed it